### DISCUSSION TITLE
Knex: Timeout when enabling elasticsearch

### DISCUSSION THREAD

**Discussion by manu-rivas** [requarks/wiki#5235] [Error / Bug Report]
Hi all,

Those are my current versions:

Wikijs version: 2.5.278
Nodejs Version: 16.14.2
Elastic Search Version: 7.9.0
Database: SqLite
Os: Linux.

Setting up elasticsearch as a search engine as follows:
![image](https://user-images.githubusercontent.com/79850103/165908966-56307cd0-048b-48c8-96cf-978ea886ba9b.png)

Will cause the error: `Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?` in wiki js, and everything stops working, forcing to restart the system.

Am I missing something?

Edit: The error occurs when rebuilding index.

Thanks in advance.

**Comment by JamesHarley** [comment]
I'm having the same issue when rebuilding the index. It freezes and I'm forced to restart the node server. Saving wiki pages one at a time does work in adding them to the index :/

**Comment by BloodshotPico** [comment]
I've got this issue right off the bat from booting up the dockers.

**Comment by boristian** [comment]
Hi,
i'm having the same issue.

When rebuilding the search index i get `warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?`.
There are about 140 pages in the wiki.

Wiki.js: 2.5.296
node.js: 16.17.1
elasticsearch: 7.17.9
Database: sqlite

Thanks in advance

**Comment by bbbart** [comment]
I'd just like to chime in that the exact same seems to be happening here as well.

Good to note perhaps is that I first started with `Database - Basic` search, and then, after creating a handful of pages, enabled `Elasticsearch` as a search engine.

wiki.js: 2.5.298
node.js: 19.8.1
elasticsearch: 7.17.9
database: SQLite

**Reply by bbbart** [reply]
(I just read that odd-numbered node versions are not supported, but after downgrading to 18.15.0 the problem persists)

**Comment by corsmond** [comment]
Same problem here. Why has an issue not been created for this?

**Comment by prohtex** [comment]
Same for me. I started with a sqlite database and am now stuck with the "basic search" that just enables us to search page titles. To wit:

- No migration path from sqlite to postgresql so can't enable real search
- AWS CloudSearch is depreciated as of June 2024 with no way to create new domains
- Elasticsearch appears completely broken

Am I missing something @NGPixel? Really great for your excellent work on this application, and hopeful that someone in my situation can enable a workable search engine for content indexing.
